tests/test_calc_barycentric_corr.py

```diff
@@ -0,0 +1,247 @@
+import unittest
+
+import astropy.wcs
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.time import Time
+import math
+
+import kbmod
+import kbmod.analysis_utils
+import kbmod.search
+
+
+class test_calc_barycentric_corr(unittest.TestCase):
+    def setUp(self):
+        # Define the path for the data.
+        # TODO: this should be ../data/demo but vscode does not get the message all the time.
+        # from time import sleep
+        # sleep(300)
+        im_filepath = "../data/demo"
+        # HACK: workaround for vscode not honoring the cwd specification.
+        from os.path import isdir
+
+        if not isdir(im_filepath):
+            im_filepath = "data/demo"
+        # The demo data has an object moving at x_v=10 px/day
+        # and y_v = 0 px/day. So we search velocities [0, 20]
+        # and angles [-0.5, 0.5].
+        v_arr = [0, 20, 21]
+        ang_arr = [0.5, 0.5, 11]
+
+        self.input_parameters = {
+            # Required
+            "im_filepath": im_filepath,
+            "res_filepath": None,
+            "time_file": None,
+            "output_suffix": "DEMO",
+            "v_arr": v_arr,
+            "ang_arr": ang_arr,
+            "bary_dist": 50.0,
+        }
+
+    def _expected_string(self, baryCoeff, shape_coeff):
+        """Return a string of the actual results in a form that can be copied into baryExpected in the test code. Used for as-built testing."""
+        value_string = f"# fmt: off\nbaryExpected = np.array(["
+        value_string += f"\n    #       dx,                      dxdx,                    dxdy,                    dy,                      dydx,                    dydy,"
+        for i in range(0, shape_coeff[0]):
+            value_string += f"\n    [{baryCoeff[i,0]:23.16f}"
+            for j in range(1, shape_coeff[1]):
+                value_string += f", {baryCoeff[i,j]:23.16f}"
+            value_string += f"],"
+        value_string += f"\n]\n# fmt: on"
+        return value_string
+
+    def _exception_strings(self, baryCoeff, baryExpected, shape_coeff):
+        """Return a list of strings of the actual results that violate the expected results and a boolean that is True if all the results are as expected."""
+        exception_strings = []
+        check_ok = True
```

Comment:
Technically you don't need `check_ok` here or below. If `exception_strings` is non-empty, then things are not okay.

Reply:
i will change